packages/cli/src/commonlib/codeFlowLogin.ts

```diff
@@ -59,7 +61,11 @@ export class CodeFlowLogin {
     }
   }
 
-  async login(): Promise<string> {
+  async login(tenantId?: string): Promise<string> {
+    if (tenantId) {
+      this.config!.auth.authority = env.activeDirectoryEndpointUrl + tenantId;
+      this.pca = new PublicClientApplication(this.config!);
```

Comment:
Change `this.pca` outside of constructor is not a good idea.
How about adding a new class `codeFlowTenantLogin` to handle tenant case only?
Since this flow will be shared by M365 and azure, better to not change the current class.

Reply:
Seperate tenant login from codeFlow.